### ISSUE TITLE
[unet_tf2] docker run --runtime=nvidia -it --gpus all --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 --rm --ipc=host -v ${PWD}/data:/data -v ${PWD}/results:/results unet_tf2:latest /bin/bash  fail

### ISSUE THREAD

**Issue by 14asaf** [NVIDIA/DeepLearningExamples#679]
I try to use in this guide:

https://github.com/NVIDIA/DeepLearningExamples/tree/master/TensorFlow2/Segmentation/UNet_Medical

I tried to run this command in stage 5:

` bash examples/unet_1GPU.sh /data /results
`
but I got  the following error:
```
 GPU (device: 0, name: A100-SXM4-40GB, pci bus id: 0000:00:03.0, compute capability: 8.0)
[1,0]<stderr>:WARNING:tensorflow:The operation `tf.image.convert_image_dtype` will be skipped since the input and output dtypes are identical.
[1,0]<stderr>:2020-09-08 10:37:13.553345: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.8
[1,0]<stderr>:2020-09-08 10:37:14.678041: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.11
[1,0]<stderr>:2020-09-08 10:37:28.079393: W tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc:95] Unknown compute capability (8, 0) .Defaulting to telling LLVM that we're compiling for sm_75
[1,0]<stderr>:2020-09-08 10:37:31.827601: I tensorflow/compiler/jit/xla_compilation_cache.cc:241] Compiled cluster using XLA!  This line is logged at most once for the lifetime of the process.
[1,0]<stderr>:Traceback (most recent call last):
[1,0]<stderr>:  File "main.py", line 69, in <module>
[1,0]<stderr>:    main()
[1,0]<stderr>:  File "main.py", line 57, in main
[1,0]<stderr>:    train(params, model, dataset, logger)
[1,0]<stderr>:  File "/workspace/unet/run.py", line 86, in train
[1,0]<stderr>:    train_step(images, labels, warmup_batch=iteration == 0)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/def_function.py", line 580, in __call__
[1,0]<stderr>:    result = self._call(*args, **kwds)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/def_function.py", line 644, in _call
[1,0]<stderr>:    return self._stateless_fn(*args, **kwds)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 2420, in __call__
[1,0]<stderr>:    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 1665, in _filtered_call
[1,0]<stderr>:    self.captured_inputs)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 1746, in _call_flat
[1,0]<stderr>:    ctx, args, cancellation_manager=cancellation_manager))
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 598, in call
[1,0]<stderr>:    ctx=ctx)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/execute.py", line 60, in quick_execute
[1,0]<stderr>:    inputs, attrs, num_outputs)
[1,0]<stderr>:tensorflow.python.framework.errors_impl.UnknownError:  ncclCommInitRank failed: unhandled system error
[1,0]<stderr>:	 [[node HorovodBroadcast_Adam_unet_output_block_conv2d_6_bias_m_0 (defined at <string>:234) ]] [Op:__inference_train_step_4208]
[1,0]<stderr>:
[1,0]<stderr>:Function call stack:
[1,0]<stderr>:train_step
[1,0]<stderr>:

```

**Environment**

* Container version : unet_tf2
* GPUs in the system: 4 x A100-SXM4-40GB:
* CUDA driver version : 450.51.06

**Comment by mmarcinkiewicz** [comment]
Hi @14asaf , thanks for reporting the issue . We are looking into that.

**Comment by 14asaf** [comment]
I followed what they were told here to do

![image](https://user-images.githubusercontent.com/39865699/93212575-6f5abb80-f76b-11ea-864c-0755683153b6.png)

**Comment by mmarcinkiewicz** [comment]
Thanks for the info.

Could you replace the line in
https://github.com/NVIDIA/DeepLearningExamples/blob/master/TensorFlow2/Segmentation/UNet_Medical/Dockerfile#L1

with
ARG FROM_IMAGE_NAME=nvcr.io/nvidia/tensorflow:20.08-tf2-py3

and give it a shot?

**Comment by 14asaf** [comment]
I changed but still got an error message

> root@08c87d449076:/workspace/unet# bash examples/unet_1GPU.sh /data /results
2020-09-15 16:40:47.100098: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
[1,0]<stderr>:2020-09-15 16:40:50.561570: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
[1,0]<stderr>:2020-09-15 16:40:51.852017: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
[1,0]<stderr>:2020-09-15 16:41:02.381823: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
[1,0]<stderr>:pciBusID: 0000:00:03.0 name: A100-SXM4-40GB computeCapability: 8.0
[1,0]<stderr>:coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
[1,0]<stderr>:2020-09-15 16:41:02.384112: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 1 with properties: 
[1,0]<stderr>:pciBusID: 0000:00:04.0 name: A100-SXM4-40GB computeCapability: 8.0
[1,0]<stderr>:coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
[1,0]<stderr>:2020-09-15 16:41:02.386346: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 2 with properties: 
[1,0]<stderr>:pciBusID: 0000:00:05.0 name: A100-SXM4-40GB computeCapability: 8.0
[1,0]<stderr>:coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
[1,0]<stderr>:2020-09-15 16:41:02.388297: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 3 with properties: 
[1,0]<stderr>:pciBusID: 0000:00:06.0 name: A100-SXM4-40GB computeCapability: 8.0
[1,0]<stderr>:coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
[1,0]<stderr>:2020-09-15 16:41:02.388342: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
[1,0]<stderr>:2020-09-15 16:41:02.419809: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.11
[1,0]<stderr>:2020-09-15 16:41:02.435910: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
[1,0]<stderr>:2020-09-15 16:41:02.444065: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
[1,0]<stderr>:2020-09-15 16:41:02.468419: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
[1,0]<stderr>:2020-09-15 16:41:02.475791: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.11
[1,0]<stderr>:2020-09-15 16:41:02.476032: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.8
[1,0]<stderr>:2020-09-15 16:41:02.492645: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0, 1, 2, 3
[1,0]<stderr>:2020-09-15 16:41:02.762780: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2200190000 Hz
[1,0]<stderr>:2020-09-15 16:41:02.766032: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7fe0ac000b20 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
[1,0]<stderr>:2020-09-15 16:41:02.766066: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
[1,0]<stderr>:2020-09-15 16:41:02.909588: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7fde48000b20 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
[1,0]<stderr>:2020-09-15 16:41:02.909626: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): A100-SXM4-40GB, Compute Capability 8.0
[1,0]<stderr>:2020-09-15 16:41:02.913386: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
[1,0]<stderr>:pciBusID: 0000:00:03.0 name: A100-SXM4-40GB computeCapability: 8.0
[1,0]<stderr>:coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
[1,0]<stderr>:2020-09-15 16:41:02.913446: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
[1,0]<stderr>:2020-09-15 16:41:02.913513: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.11
[1,0]<stderr>:2020-09-15 16:41:02.913537: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
[1,0]<stderr>:2020-09-15 16:41:02.913552: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
[1,0]<stderr>:2020-09-15 16:41:02.913563: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
[1,0]<stderr>:2020-09-15 16:41:02.913576: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.11
[1,0]<stderr>:2020-09-15 16:41:02.913586: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.8
[1,0]<stderr>:2020-09-15 16:41:02.917963: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0
[1,0]<stderr>:2020-09-15 16:41:02.919126: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.11.0
[1,0]<stderr>:2020-09-15 16:41:05.002134: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:
[1,0]<stderr>:2020-09-15 16:41:05.002193: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 
[1,0]<stderr>:2020-09-15 16:41:05.002205: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N 
[1,0]<stderr>:2020-09-15 16:41:05.008347: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 37604 MB memory) -> physical GPU (device: 0, name: A100-SXM4-40GB, pci bus id: 0000:00:03.0, compute capability: 8.0)
[1,0]<stderr>:WARNING:tensorflow:The operation `tf.image.convert_image_dtype` will be skipped since the input and output dtypes are identical.
[1,0]<stderr>:2020-09-15 16:41:10.954317: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.8
[1,0]<stderr>:2020-09-15 16:41:12.501689: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.11
[1,0]<stderr>:2020-09-15 16:41:28.345060: W tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc:95] Unknown compute capability (8, 0) .Defaulting to telling LLVM that we're compiling for sm_75
[1,0]<stderr>:2020-09-15 16:41:32.183132: I tensorflow/compiler/jit/xla_compilation_cache.cc:241] Compiled cluster using XLA!  This line is logged at most once for the lifetime of the process.
[1,0]<stderr>:Traceback (most recent call last):
[1,0]<stderr>:  File "main.py", line 69, in <module>
[1,0]<stderr>:    main()
[1,0]<stderr>:  File "main.py", line 57, in main
[1,0]<stderr>:    train(params, model, dataset, logger)
[1,0]<stderr>:  File "/workspace/unet/run.py", line 86, in train
[1,0]<stderr>:    train_step(images, labels, warmup_batch=iteration == 0)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/def_function.py", line 580, in __call__
[1,0]<stderr>:    result = self._call(*args, **kwds)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/def_function.py", line 644, in _call
[1,0]<stderr>:    return self._stateless_fn(*args, **kwds)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 2420, in __call__
[1,0]<stderr>:    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 1665, in _filtered_call
[1,0]<stderr>:    self.captured_inputs)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 1746, in _call_flat
[1,0]<stderr>:    ctx, args, cancellation_manager=cancellation_manager))
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/function.py", line 598, in call
[1,0]<stderr>:    ctx=ctx)
[1,0]<stderr>:  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/execute.py", line 60, in quick_execute
[1,0]<stderr>:    inputs, attrs, num_outputs)
[1,0]<stderr>:tensorflow.python.framework.errors_impl.UnknownError:  ncclCommInitRank failed: unhandled system error
[1,0]<stderr>:	 [[node HorovodBroadcast_Adam_unet_bottleneck_block_conv2d_transpose_bias_v_0 (defined at <string>:234) ]] [Op:__inference_train_step_4208]
[1,0]<stderr>:
[1,0]<stderr>:Function call stack:
[1,0]<stderr>:train_step
[1,0]<stderr>:

> --------------------------------------------------------------------------
> Primary job  terminated normally, but 1 process returned
> a non-zero exit code. Per user-direction, the job has been aborted.
> --------------------------------------------------------------------------
> --------------------------------------------------------------------------

> 
> mpirun detected that one or more processes exited with non-zero status, thus causing
> the job to be terminated. The first process to do so was:
> 

> 
>   Process name: [[5486,1],0]
>   Exit code:    1
> --------------------------------------------------------------------------



![image](https://user-images.githubusercontent.com/39865699/93239885-12233200-f78c-11ea-8bae-27acb15381f8.png)

**Comment by mmarcinkiewicz** [comment]
The log looks as expected until the error. I will try to reproduce the issue.

Can you try two more things?
1. Run it on a 1-gpu instance (the same commands)
2. Run:
`echo $NCCL_VERSION`
`export NCCL_DEBUG=INFO` and then 
 `bash examples/unet_1GPU.sh /data /results`

and paste here the complete log, as you have done so above.

**Comment by 14asaf** [comment]
`1.Run it on a 1-gpu instance (the same commands)`

I'm working on Google's cloud and I'm unable to change the number of gpus on this instance,
therefore, I did it manual by:

`export CUDA_VISIBLE_DEVICES=1`

but still I got the error


2.  I attached the log file
[output.log](https://github.com/NVIDIA/DeepLearningExamples/files/5230834/output.log)

**Comment by mmarcinkiewicz** [comment]
Thanks @14asaf.

I'll look into that and come back to you.

**Comment by mmarcinkiewicz** [comment]
It looks like a bug in NCCL 2.7.5, which should have been fixed in 2.7.8. The 20.08 container I pointed you before should have NCCL 2.7.8. May I ask you to repeat the procedure of collecting logs but using the 20.08 container?

Meaning:
1. Change the Dockerfile to contain `ARG FROM_IMAGE_NAME=nvcr.io/nvidia/tensorflow:20.08-tf2-py3`
2. Build and run the container
3. Run `export NCCL_DEBUG=INFO`
4. Run `bash examples/unet_1GPU.sh /data /results`
